src/psyclone/psyir/nodes/read_only_verify_node.py

```diff
@@ -0,0 +1,157 @@
+        # and post-variables (i.e. here input and output). A variable
+        # "myvar" will be stored as "myvar" with its input value, and
+        # "myvar_post" with its output value.
+        self._post_name = "_readonlyverify"
+
+        # Store the list of input- and output-variables, so that a driver
+        # generator can get the list of variables that are written.
+        self._input_list = []
+        self._output_list = []
+
+    @property
+    def read_only_verify_body(self):
+        '''
+        :returns: the Schedule associated with this ExtractNode.
+        :rtype: :py:class:`psyclone.psyir.nodes.Schedule`
+
+        '''
+        return super(ReadOnlyVerifyNode, self).psy_data_body
+
+    @property
+    def dag_name(self):
+        '''
+        Returns the name to use in a DAG for this Node
+
+        :returns: the dag name of ExtractNode.
+        :rtype: str
+        '''
+        return "read_only_verify_" + str(self.position)
+
+    def update_vars_and_postname(self):
+        '''
+        This function is called after the variables to be extracted
+        have been stored in self._input_list and self._output_list.
+        It can be used to e.g. remove unnecessary variables (e.g. loop
+        counter), or adjust the postfix to assure that no duplicated
+        variable name is created. This default function does not
+        do anything atm.
+        '''
+
+    def gen_code(self, parent):
+        # pylint: disable=arguments-differ
+        '''
+        Generates the code required for extraction of one or more Nodes.
+        It uses the PSyData API (via the base class PSyDataNode) to create
+        the required callbacks that will allow a library to write the
+        kernel data to a file.
+
+        :param parent: the parent of this Node in the PSyIR.
+        :type parent: :py:class:`psyclone.psyir.nodes.Node`.
+        '''
+
+        # Determine the variables to write:
+        from psyclone.core.access_info import VariablesAccessInfo
+        variables_info = VariablesAccessInfo(self)
+        read_only = []
+        for var_name in variables_info:
+            if variables_info[var_name].is_read_only():
+                read_only.append(var_name)
+
+        # Add a callback here so that derived classes can adjust the list
+        # of variables to provide, or the suffix used (which might
+        # depend on the variable name which could create clashes).
+        self.update_vars_and_postname()
+
+        options = {'pre_var_list': read_only,
+                   'post_var_list': read_only}
+
+        from psyclone.f2pygen import CommentGen
+        parent.add(CommentGen(parent, ""))
+        parent.add(CommentGen(parent, " ReadOnlyVerifyStart"))
+        parent.add(CommentGen(parent, ""))
+        super(ReadOnlyVerifyNode, self).gen_code(parent, options)
+        parent.add(CommentGen(parent, ""))
+        parent.add(CommentGen(parent, " ReadOnlyVerifyEnd"))
+        parent.add(CommentGen(parent, ""))
```

Comment:
Please could you add an `__all__` at the end of this class.

Reply:
Done.